src/picongpu/include/particles/Particles.tpp

```diff
@@ -72,7 +73,9 @@ datasetID( datasetID )
 
     log<picLog::MEMORY > ( "size for all exchange = %1% MiB" ) % ( (double) sizeOfExchanges / 1024. / 1024. );
 
-    const uint32_t commTag = FrameType::CommunicationTag + SPECIES_FIRSTTAG;
+    const uint32_t commTag = PMacc::traits::GetUniqueTypeId<FrameType>::uid + SPECIES_FIRSTTAG;
```

Comment:
The unique id begins with 0 (zero) and is increased with each static variable in the code. All ids are smaller than `2^15-1`.
I can add a test in GridBuffer or in the MPI Layer that the tag is valid.